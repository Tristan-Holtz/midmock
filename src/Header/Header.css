form {
  display: flex;
  justify-content: space-around;
  width: 100%;
}

header {
  margin-bottom: 2em;
  width: 100%;
}

h1 {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

.reservation-button {
  border: 2px solid #fff;
  background-color: teal;
  font-size: 1em;
  color: #fff;
  width: 15em;
}

input {
  height: 3em;
  font-size: 16px;
}
