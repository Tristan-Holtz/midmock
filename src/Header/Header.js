import React, { Component } from 'react';
import './Header.css';

class Header extends Component {
  constructor() {
    super();
    this.state = {
      id: Date.now(),
      name: '',
      date: '',
      time: '',
      number: ''
    };
  }

  makeReservation = () => {
    const options = {
      method: 'POST',
      body: JSON.stringify(this.state),
      headers: {
        'Content-Type': 'application/json'
      }
    };

    fetch('http://localhost:3001/api/v1/reservations', options)
      .then(response => response.json())
      .then(data => console.log(data))
      .catch(error => console.log(error));
  };

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    return (
      <header>
        <h1>Turing Cafe Reservations</h1>
        <form>
          <input
            placeholder="Name..."
            type="text"
            name="name"
            onChange={this.handleChange}
          ></input>
          <input
            placeholder="Date (MM/DD)"
            type="text"
            name="date"
            onChange={this.handleChange}
          ></input>
          <input
            placeholder="Time"
            type="text"
            name="time"
            onChange={this.handleChange}
          ></input>
          <input
            placeholder="Number of guests"
            type="text"
            name="number"
            onChange={this.handleChange}
          ></input>
          <button onClick={this.makeReservation} className="reservation-button">
            Make Reservation
          </button>
        </form>
      </header>
    );
  }
}

export default Header;
