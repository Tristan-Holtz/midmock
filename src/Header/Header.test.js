import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('should render the component', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update name in state', () => {
      const wrapper = shallow(<Header />);
      const mockEvent = {
        target: {
          name: 'name',
          value: 'Tristan'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual('Tristan');
    });

    it('should update date in state', () => {
      const wrapper = shallow(<Header />);
      const mockEvent = {
        target: {
          name: 'date',
          value: '2/14'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('date')).toEqual('2/14');
    });

    it('should update time in state', () => {
      const wrapper = shallow(<Header />);
      const mockEvent = {
        target: {
          name: 'time',
          value: '7:30'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('time')).toEqual('7:30');
    });

    it('should update number in state', () => {
      const wrapper = shallow(<Header />);
      const mockEvent = {
        target: {
          name: 'number',
          value: '2'
        }
      };

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('number')).toEqual('2');
    });
  });

  describe('makeReservation', () => {
    it('should fire when button is clicked', () => {
      const wrapper = shallow(<Header />);
      wrapper.makeReservation = jest.fn();

      wrapper.find('button').simulate('click');
      expect(wrapper.makeReservation).toHaveBeenCalled();
    });
  });
});
