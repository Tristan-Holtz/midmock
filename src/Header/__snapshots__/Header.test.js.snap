// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<header>
  <h1>
    Turing Cafe Reservations
  </h1>
  <form>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name..."
      type="text"
    />
    <input
      name="date"
      onChange={[Function]}
      placeholder="Date (MM/DD)"
      type="text"
    />
    <input
      name="time"
      onChange={[Function]}
      placeholder="Time"
      type="text"
    />
    <input
      name="number"
      onChange={[Function]}
      placeholder="Number of guests"
      type="text"
    />
    <button
      className="reservation-button"
      onClick={[Function]}
    >
      Make Reservation
    </button>
  </form>
</header>
`;
